Building PDMS w/clang

## Current Behavior

Build fails due to decrementing a std::unordered_set.

```cpp
PdmsTools.cpp:212:55: error: cannot decrement value of type 'ElementsStack::iterator' (aka '__hash_const_iterator<std::__1::__hash_node<PdmsTools::PdmsObjects::GenericItem *, void *> *>')
                ElementsStack::iterator it = s_elementsStack.end(); --it;
```

(in 2 places)

## Expected Behavior

Should compile :-)

## Context/Comments

According to the standard, a std::unordered_set iterator is _at least_ a forward iterator. Seems like clang has decided to go with that while other compilers have chosen to make it bidirectional.

More info here:

   http://www.cplusplus.com/reference/unordered_set/unordered_set/

## Your Environment
<!--- Include any relevant details about your environment -->
* OS & Version: macOS 10.12.6
* CC Version: 2.9 git
* Compiler: Apple LLVM version 8.1.0 (clang-802.0.42)

--- a/libs/qCC_io/PDMS/PdmsTools.cpp
+++ b/libs/qCC_io/PDMS/PdmsTools.cpp
@@ -15,15 +15,15 @@
 //#                                                                        #
 //##########################################################################
 
+#include <QSet>
+
 #include "PdmsTools.h"
 
 //System
-#include <string.h>
+#include <assert.h>
 #include <iostream>
 #include <stdlib.h>
-#include <assert.h>
-#include <unordered_set>
-#include <iterator>
+#include <string.h>
 
 //qCC_db
 //#include <ccLog.h>
@@ -47,8 +47,8 @@ static GroupElement defaultWorld(PDMS_WORLD);
 ///////////////////////////////
 // ITEM STACK
 ///////////////////////////////
-typedef std::unordered_set<PdmsObjects::GenericItem*> ElementsStack;
-ElementsStack s_elementsStack;
+typedef QSet<PdmsObjects::GenericItem*> ElementsStack;
+static ElementsStack s_elementsStack;
 
 void PdmsObjects::Stack::Init()
 {
@@ -62,27 +62,26 @@ void PdmsObjects::Stack::Clear()
 	while (!s_elementsStack.empty())
 	{
 		GenericItem* item = *(s_elementsStack.begin());
-		s_elementsStack.erase(item);
+		
+		s_elementsStack.remove( item );
+		
 		if (item)
 		{
 			//ccLog::Print(QString("[PDMS] Should be deleting %1").arg(item->name));
 			delete item;
 		}
 	}
+	
 	s_elementsStack.clear();
 }
 
 void PdmsObjects::Stack::Destroy(GenericItem* &item)
 {
-	if (item)
+	if ( item && s_elementsStack.remove( item ) )
 	{
 		//ccLog::Print(QString("[PDMS] Destroying %1").arg((item)->name));
-		if (s_elementsStack.find(item) != s_elementsStack.end())
-		{
-			s_elementsStack.erase(item);
-			delete item;
-		}
-		item = 0;
+		delete item;
+		item = nullptr;
 	}
 }
 

